src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/service/PrisonerValidationService.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.prison.visitbooker.registry.service
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.PermittedPrisonerDto
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.enums.PrisonerValidationErrorCodes
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.enums.PrisonerValidationErrorCodes.PRISONER_RELEASED
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.enums.PrisonerValidationErrorCodes.PRISONER_TRANSFERRED
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.prisoner.search.PrisonerDto
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.exception.PrisonerValidationException
+
+@Service
+class PrisonerValidationService(
+  private val bookerDetailsService: BookerDetailsService,
+  private val prisonerSearchService: PrisonerSearchService,
+) {
+  private companion object {
+    private val LOG = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun validatePrisoner(bookerReference: String, prisonerId: String) {
+    LOG.info("Validate booking by booker - {} for prisoner - {}", bookerReference, prisonerId)
+
+    val permittedPrisoner = bookerDetailsService.getPermittedPrisoner(bookerReference, prisonerId).let { permittedPrisoner ->
+      PermittedPrisonerDto(
```

Comment:
This Dto takes a PermittedPrisoner as a constructor value. Could you do this:

.let { permittedPrisonerEntity -> PermittedPrisonerDto(permittedPrisonerEntity) }

or even

val permittedPrisoner = PermittedPrisonerDto(bookerDetailsService.getPermittedPrisoner(bookerReference, prisonerId))

Reply:
this is giving an error as the PermittedPrisonerDto() constructor call tries to lazily load the visitors foe a prisoner. As discussed, not updated but will look at ways to improve the constructor call later